storage/client_test.go

```diff
@@ -379,178 +373,3 @@ func internalTruncateLogArgs(index uint64, raftID int64, storeID proto.StoreID)
-		left := &proto.RangeDescriptor{RaftID: int64(i * 2), StartKey: test.leftStart, EndKey: test.leftEnd}
-		right := &proto.RangeDescriptor{RaftID: int64(i*2 + 1), StartKey: test.rightStart, EndKey: test.rightEnd}
-		var calls []client.Call
-		if test.split {
-			var err error
-			if calls, err = storage.SplitRangeAddressing(left, right); err != nil {
-				t.Fatal(err)
-			}
-		} else {
-			var err error
-			if calls, err = storage.MergeRangeAddressing(left, right); err != nil {
-				t.Fatal(err)
-			}
-		}
-		if err := store.DB().Run(calls...); err != nil {
-			t.Fatal(err)
-		}
-		// Scan meta keys directly from engine.
-		kvs, err := engine.MVCCScan(store.Engine(), engine.KeyMetaPrefix, engine.KeyMetaMax, 0, proto.MaxTimestamp, true, nil)
-		if err != nil {
-			t.Fatal(err)
-		}
-		metas := metaSlice{}
-		for _, kv := range kvs {
-			scannedDesc := &proto.RangeDescriptor{}
-			if err := gogoproto.Unmarshal(kv.Value.Bytes, scannedDesc); err != nil {
-				t.Fatal(err)
-			}
-			metas = append(metas, metaRecord{key: kv.Key, desc: scannedDesc})
-		}
-
-		// Continue to build up the expected metas slice, replacing any earlier
-		// version of same key.
-		addOrRemoveNew := func(keys []proto.Key, desc *proto.RangeDescriptor, add bool) {
-			for _, n := range keys {
-				found := -1
-				for i := range expMetas {
-					if expMetas[i].key.Equal(n) {
-						found = i
-						expMetas[i].desc = desc
-						break
-					}
-				}
-				if found == -1 && add {
-					expMetas = append(expMetas, metaRecord{key: n, desc: desc})
-				} else if found != -1 && !add {
-					expMetas = append(expMetas[:found], expMetas[found+1:]...)
-				}
-			}
-		}
-		addOrRemoveNew(test.leftExpNew, left, test.split /* on split, add; on merge, remove */)
-		addOrRemoveNew(test.rightExpNew, right, true)
-		sort.Sort(expMetas)
-
-		if test.split {
-			log.V(1).Infof("test case %d: split %q-%q at %q", i, left.StartKey, right.EndKey, left.EndKey)
-		} else {
-			log.V(1).Infof("test case %d: merge %q-%q + %q-%q", i, left.StartKey, left.EndKey, left.EndKey, right.EndKey)
-		}
-		for _, meta := range metas {
-			log.V(1).Infof("%q", meta.key)
-		}
-		log.V(1).Infof("")
-
-		if !reflect.DeepEqual(expMetas, metas) {
-			t.Errorf("expected metas don't match")
-			if len(expMetas) != len(metas) {
-				t.Errorf("len(expMetas) != len(metas); %d != %d", len(expMetas), len(metas))
-			} else {
-				for i, meta := range expMetas {
-					if !meta.key.Equal(metas[i].key) {
-						fmt.Printf("%d: expected %q vs %q\n", i, meta.key, metas[i].key)
-					}
-					if !reflect.DeepEqual(meta.desc, metas[i].desc) {
-						fmt.Printf("%d: expected %q vs %q and %s vs %s\n", i, meta.key, metas[i].key, meta.desc, metas[i].desc)
-					}
-				}
-			}
-		}
-	}
-}
-
-// TestUpdateRangeAddressingSplitMeta1 verifies that it's an error to
-// attempt to update range addressing records that would allow a split
-// of meta1 records.
-func TestUpdateRangeAddressingSplitMeta1(t *testing.T) {
```

Comment:
Do we have other tests that cover what these tests did? If these tests were useful we should keep them even if it means some extra exported functions.